.github/workflows/post-release.yml

```diff
@@ -0,0 +1,92 @@
+name: Post Release
+on:
+  workflow_dispatch:
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  promote-image:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+          
+      - name: Install kpromo
+        run: go install sigs.k8s.io/promo-tools/v4/cmd/kpromo@latest
+        
+      - name: Get latest pre-release tag
+        id: extract_tag
+        run: |
+          TAG=$(gh release list --limit 1 --json tagName,isPrerelease --jq '.[] | select(.isPrerelease == true) | .tagName')
+          if [ -z "$TAG" ]; then
+            echo "Error: No pre-release found"
+            exit 1
+          fi
+          echo "tag=${TAG}" >> $GITHUB_OUTPUT
+          echo "Found pre-release tag: ${TAG}"
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        
+      - name: Create image promotion PR
+        run: |
+          kpromo pr --fork=${{ github.actor }} -i \
+            --project=kube-state-metrics \
+            --tag=${{ steps.extract_tag.outputs.tag }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Hmm, I only see the second part of this workflow in the [test run](https://github.com/Rishab87/kube-state-metrics/actions/runs/18229006152/job/51907490769). Were you able to test this out on your local?

Reply:
Does the scope of the GITHUB_TOKEN work outside of its own repository for PRs?